### PR TITLE
Add some more tensor docs

### PR DESCRIPTION
**Pull request by dsyme**
Starting to populate docs in Tensor.fs

Also testing that math can be embedded in docs (see `arange` docs)

There are some issues with cross-references, I'll work on those tomorrow, also will do https://github.com/fsprojects/FSharp.Formatting/issues/604

### CODE REVIEW

File: src/DiffSharp.Core/Tensor.fs [hunk 1]
```diff
@@ -265,6 +284,7 @@ type Tensor =
             let fields = c.GetFields()
             sprintf "TensorR %A %s" t.shape c.Name
 
+    /// A debugging routine to compute the parents of a tensor involved in reverse-mode autmoatic differentiation
```
Comment (kevmal) [inline src/DiffSharp.Core/Tensor.fs:287]:
```suggestion
    /// A debugging routine to compute the parents of a tensor involved in reverse-mode automatic differentiation

```

File: src/DiffSharp.Core/Tensor.fs [hunk 2]
```diff
@@ -340,67 +380,147 @@ type Tensor =
...
         override t.ToUInt64(_) = failwithf "Cannot convert Tensor to UInt64"
 
+    /// Indicates if two tensors have the same shape and all corresponding elements are equal within the
+    /// given tolerances.
     member t.allclose(tensor:Tensor, ?relativeTolerance, ?absoluteTolerance) =
         let relativeTolerance = defaultArg relativeTolerance 1e-5
         let absoluteTolerance = defaultArg absoluteTolerance 1e-8
         t.primalRaw.AllClose(tensor.primalRaw, relativeTolerance, absoluteTolerance)
 
+    /// Returns a new tensor filled with '1' values for the given shape, element type and configuration, defaulting to the
```
Comment (kevmal) [inline src/DiffSharp.Core/Tensor.fs:390]:
```suggestion
    /// Returns a new tensor filled with '0' values for the given shape, element type and configuration, defaulting to the 

```

File: src/DiffSharp.Core/Tensor.fs [hunk 3]
```diff
@@ -340,67 +380,147 @@ type Tensor =
...
     member a.scalarLike(scalar:IConvertible, ?dtype, ?device, ?backend) = 
         a.fullLike(scalar, [], ?dtype=dtype, ?device=device, ?backend=backend)
+
+    /// Returns a new tensor with random values drawn from the uniform distribution [0,1) for the
+    /// given shape, element type and configuration, defaulting to the shape and configuration of the input tensor.
     member a.randLike(?shape:seq<int>, ?dtype, ?device, ?backend) = 
         let shape = defaultArg shape (a.shape |> Array.toSeq)
         Tensor(a.primalRaw.RandomLike((shape |> Array.ofSeq), ?dtype=dtype, ?device=device, ?backend=backend))
+
+    /// Returns a new tensor with random values drawn from the normal distribution, for the
```
Comment (kevmal) [inline src/DiffSharp.Core/Tensor.fs:419]:
```suggestion
    /// Returns a new tensor with random values drawn from the standard normal distribution, for the

```

**Review by kevmal — CHANGES_REQUESTED** [review]

**Comment by dsyme** [comment]
@kevmal Thanks for the proof reading!

**Comment by dsyme** [comment]
The codecov on the patch failed but it's only documentation, so will merge

**Pull request merged**